kolibri/core/assets/src/vue/icon-button.vue

```diff
@@ -29,7 +29,9 @@
     computed: {
       hasIcon() {
         // something of a hack but seems to work fine
-        return this.$el.querySelector('svg');
+        return this.$nextTick(() => {
+          this.$el.querySelector('svg');
+        });
```

Comment:
At one point we tried to do this using the `v-ref` attribute, but it turned out that you couldn't use that with `slot`s (https://github.com/vuejs/vue/issues/2128).

It was also not possible in vue1 to access slots programmatically, but it sounds like that's now possible in vue2:

* https://github.com/vuejs/vue/issues/1726
* https://vuejs.org/v2/api/#vm-slots

I would check whether it's possible to use the new `$slots` attribute to query for the existence of an icon rather than using the hacky `querySelector` logic.